internal/resource/config/localidentity/identityprofile/local_identity_identity_profile_common.go

```diff
@@ -0,0 +1,176 @@
+		"email_verification_sent_template_name":    basetypes.StringType{},
+		"email_verification_success_template_name": basetypes.StringType{},
+		"email_verification_error_template_name":   basetypes.StringType{},
+		"email_verification_type":                  basetypes.StringType{},
+		"otp_length":                               basetypes.Int64Type{},
+		"otp_retry_attempts":                       basetypes.Int64Type{},
+		"allowed_otp_character_set":                basetypes.StringType{},
+		"otp_time_to_live":                         basetypes.Int64Type{},
+		"email_verification_otp_template_name":     basetypes.StringType{},
+		"otl_time_to_live":                         basetypes.Int64Type{},
+		"field_for_email_to_verify":                basetypes.StringType{},
+		"field_storing_verification_status":        basetypes.StringType{},
+		"notification_publisher_ref":               basetypes.ObjectType{AttrTypes: resourcelink.AttrType()},
+		"require_verified_email":                   basetypes.BoolType{},
+		"require_verified_email_template_name":     basetypes.StringType{},
+	}
+
+	dsConfigAttrTypes = map[string]attr.Type{
+		"base_dn":                  basetypes.StringType{},
+		"type":                     basetypes.StringType{},
+		"data_store_ref":           basetypes.ObjectType{AttrTypes: resourcelink.AttrType()},
+		"data_store_mapping":       basetypes.MapType{ElemType: types.ObjectType{AttrTypes: dsMappingAttrtypes}},
+		"create_pattern":           basetypes.StringType{},
+		"object_class":             basetypes.StringType{},
+		"auxiliary_object_classes": basetypes.SetType{ElemType: basetypes.StringType{}},
+	}
+
+	dsMappingAttrtypes = map[string]attr.Type{
+		"type":     basetypes.StringType{},
+		"name":     basetypes.StringType{},
+		"metadata": basetypes.MapType{ElemType: basetypes.StringType{}},
+	}
+)
+
+type localIdentityIdentityProfileModel struct {
+	Id                      types.String `tfsdk:"id"`
+	ProfileId               types.String `tfsdk:"profile_id"`
+	Name                    types.String `tfsdk:"name"`
+	ApcId                   types.Object `tfsdk:"apc_id"`
+	AuthSources             types.List   `tfsdk:"auth_sources"`
+	AuthSourceUpdatePolicy  types.Object `tfsdk:"auth_source_update_policy"`
+	RegistrationEnabled     types.Bool   `tfsdk:"registration_enabled"`
+	RegistrationConfig      types.Object `tfsdk:"registration_config"`
+	ProfileConfig           types.Object `tfsdk:"profile_config"`
+	FieldConfig             types.Object `tfsdk:"field_config"`
+	EmailVerificationConfig types.Object `tfsdk:"email_verification_config"`
+	DataStoreConfig         types.Object `tfsdk:"data_store_config"`
+	ProfileEnabled          types.Bool   `tfsdk:"profile_enabled"`
+}
+
+// Read a DseeCompatAdministrativeAccountResponse object into the model struct
+func readLocalIdentityIdentityProfileResponseDataSource(ctx context.Context, r *client.LocalIdentityProfile, state *localIdentityIdentityProfileModel) diag.Diagnostics {
```

Comment:
Should this be in common? I see the method still has the "DataSource" suffix, and I'm not sure it's used in the resource.